### DISCUSSION TITLE
Right mouse click as Snap Targets modifier

### DISCUSSION THREAD

**Discussion by slunak** [rxhanson/RectanglePro-Community#376] [Ideas]
Having modifier key to enable snap targets is great! This can be enhanced even further by adding an option to use right mouse click as a modifier. This way we will have keyboard-less way to use snap targets.

**Comment by rxhanson** [comment]
Thanks for the feedback. To make sure I understand, are you looking to drag the window with right mouse down? I don't think that's possible within macOS, but maybe I'm not understanding or maybe this would lead to a better idea altogether.

**Reply by slunak** [reply]
Hey, thanks for coming back!

I was referring to the below option (Only show Snap Targets when holding a modifier key):

![image](https://github.com/rxhanson/RectanglePro-Community/assets/8351121/73038ce2-639a-425b-848a-22ca0668d67a)

So my idea is instead holding a modifier key, we can hold / press right mouse button:

1. Start dragging a window (as you would normally do with left mouse button)
2. Press and hold right mouse button to view snap targets
3. Keep dragging window into your chosen snap target
4. Let the window snap by releasing both mouse buttons

Hope it is more clear now.

**Reply by asgeirtj** [reply]
Second this! This feature is available on PC with the Microsoft PowerToys Fancy Zones utility. It's great since it means you can throw windows around when your other hand is not on the keyboard. Window Throw feature allows using mouse buttons which is great but it seems that right click is not possible with that feature (and I find using scroll button or side buttons cumbersome, really bad for rsi to hold these buttons and mouse around). The feature makes sense since I don't you ever need to to move the mouse with right click down to do anything.

**Comment by asgeirtj** [comment]
Hey @slunak, just wanted to share that I discovered an app that offers a similar feature, although it's a bit less robust. The app is called BetterMouse and lets you perform actions by right-clicking and moving the mouse in a specific direction. You can check it out at https://better-mouse.com/.

By the way, @rxhanson, incorporating this feature in the powertoys style would be absolutely fantastic!

**Reply by mrapavy** [reply]
+1

I would also like to see using both mouse keys (left + right) as a window throw modifier (similar to how PowerToys/FancyZones do window snapping).
